js/privatebin.js

```diff
@@ -2504,18 +2529,24 @@ jQuery.PrivateBin = (function($, RawDeflate) {
                 return;
             }
 
-            // escape HTML entities, link URLs, sanitize
-            const escapedLinkedText = Helper.urls2links(text),
-                  sanitizedLinkedText = DOMPurify.sanitize(
-                    escapedLinkedText, {
-                        ALLOWED_TAGS: ['a'],
-                        ALLOWED_ATTR: ['href', 'rel']
-                    }
-                  );
-            $plainText.html(sanitizedLinkedText);
-            $prettyPrint.html(sanitizedLinkedText);
+            let processedText = Helper.preformatTextForDomPurify(text, format);
+
+            // link URLs
+            processedText = Helper.urls2links(processedText);
 
             switch (format) {
+                case 'syntaxhighlighting':
+                    // yes, this is really needed to initialize the environment
+                    if (typeof prettyPrint === 'function')
+                    {
+                        prettyPrint();
+                    }
+
+                    $prettyPrint.html(
+                        DOMPurify.sanitize(
+                            prettyPrintOne(processedText, null, true)
+                        )
+                    );
```

Comment:
If you want to avoid the fall-though, please add a break. It works as is, but you needlessly add the markdown version in the hidden $plainText.

I agree on the logic being convoluted with the switch, which is why I had rewritten it in d2e9e47b673f